SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestInitialOnly/SpatialTestInitialOnlyForInterestActor.cpp

```diff
@@ -0,0 +1,170 @@
+		ASpatialFunctionalTestFlowController* FlowController = GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1);
+		ATestMovementCharacter* TestCharacter =
+			GetWorld()->SpawnActor<ATestMovementCharacter>(FVector(400.0f, 0.0f, 40.0f), FRotator::ZeroRotator, FActorSpawnParameters());
+		APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
+
+		// Set a reference to the previous Pawn so that it can be processed back in the last step of the test
+		OriginalPawn = TPair<AController*, APawn*>(PlayerController, PlayerController->GetPawn());
+
+		RegisterAutoDestroyActor(TestCharacter);
+		PlayerController->Possess(TestCharacter);
+
+		FinishStep();
+	});
+
+	AddStep(TEXT("Move character to cube"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		ASpatialFunctionalTestFlowController* FlowController = GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1);
+		APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
+		ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+
+		// Move the character to the correct location
+		PlayerCharacter->SetActorLocation(FVector(-350.0f, 0.0f, 40.0f));
+
+		FinishStep();
+	});
+
+	AddStep(
+		TEXT("Check default value."), FWorkerDefinition::Client(1),
+		[this]() -> bool {
+			bool IsReady = false;
+			TArray<AActor*> SpawnActors;
+			UGameplayStatics::GetAllActorsOfClass(GetWorld(), ASpatialTestInitialOnlySpawnActor::StaticClass(), SpawnActors);
+			for (AActor* Actor : SpawnActors)
+			{
+				ASpatialTestInitialOnlySpawnActor* SpawnActor = Cast<ASpatialTestInitialOnlySpawnActor>(Actor);
+				if (SpawnActor != nullptr)
+				{
+					IsReady = true;
+				}
+			}
+			return IsReady;
+		},
+		[this]() {
+			TArray<AActor*> SpawnActors;
+			UGameplayStatics::GetAllActorsOfClass(GetWorld(), ASpatialTestInitialOnlySpawnActor::StaticClass(), SpawnActors);
+			for (AActor* Actor : SpawnActors)
+			{
+				ASpatialTestInitialOnlySpawnActor* SpawnActor = Cast<ASpatialTestInitialOnlySpawnActor>(Actor);
+				if (SpawnActor != nullptr)
+				{
+					AssertTrue(SpawnActor->Int_Initial == 1, TEXT("Check Actor.Int_Initial value."));
+					AssertTrue(SpawnActor->Int_Replicate == 1, TEXT("Check Actor.Int_Replicate value."));
+				}
+			}
+
+			FinishStep();
+		});
+
+	AddStep(TEXT("Server change value."), FWorkerDefinition::Server(1), nullptr, [this]() {
+		TArray<AActor*> SpawnActors;
+		UGameplayStatics::GetAllActorsOfClass(GetWorld(), ASpatialTestInitialOnlySpawnActor::StaticClass(), SpawnActors);
+		for (AActor* Actor : SpawnActors)
+		{
+			ASpatialTestInitialOnlySpawnActor* SpawnActor = Cast<ASpatialTestInitialOnlySpawnActor>(Actor);
+			if (SpawnActor != nullptr)
+			{
+				SpawnActor->Int_Initial = 2;
+				SpawnActor->Int_Replicate = 2;
+			}
+		}
+
+		FinishStep();
+	});
+
+	AddStep(
+		TEXT("Check changed value."), FWorkerDefinition::Client(1),
+		[this]() -> bool {
```

Comment:
This has already been checked on a previous step, and could be omitted this time.